lib/model/game_state.dart

```diff
@@ -35,6 +38,29 @@ class GameState with ChangeNotifier {
       discard.add(currentCard);
     }
 
+    // TODO - Game Result
+    switch (result){
+      case GameResult.WIN: {
+
+      }
+      break;
+
+      case GameResult.LOSE: {
+
+      }
+      break;
+
+    }
```

Comment:
I don't think the `break` statements here will do what you want - you're after `return` if you want to get out of the method completely. 

Could also be simplified to just:

```
if (result != null) {
    return;
}
```

Not sure if this is the best place for it to live - perhaps we could move it into `CardAction` and have the `performAction` method become:

```
void performAction(GameState state) {
    state.playerHand.deductResources(cost);
    performActionImpl(state);
    if (state.result == null) {
        state.nextCard():
    }
  }
```

Just feels a little neater to prevent the call to `nextCard()` - that way the `nextCard` method doesn't have any surprising behaviour which isn't obvious from the method name.

Reply:
This was mainly a way of trying to start the discussion of how to handle it. I've made the small change you mentioned for now and we can pick it up from there.